include/tscore/ink_resolver.h

```diff
@@ -157,10 +159,8 @@ enum HostResPreference {
 };
 /// # of preference values.
 static int const N_HOST_RES_PREFERENCE = HOST_RES_PREFER_IPV6 + 1;
-/// # of entries in a preference ordering.
-static int const N_HOST_RES_PREFERENCE_ORDER = 3;
 /// Storage for preference ordering.
-typedef HostResPreference HostResPreferenceOrder[N_HOST_RES_PREFERENCE_ORDER];
+using HostResPreferenceOrder = std::array<HostResPreference, 3>;
```

Comment:
Why not `N_HOST_RES_PREFERENCE` instead of `3`? Avoid explicit numbers where possible. Another alternative is to get rid of `N_HOST_RES_PREFERENCE` and put the computation for that value in place of 3, then use `.size()` as needed.